readme.md

```diff
@@ -8,23 +8,70 @@
 [![Backers][backers-badge]][collective]
 [![Chat][chat-badge]][chat]
 
-[**rehype**][rehype] plugin to automatically add `target` and `rel` attributes
-to external links.
+**[rehype][]** plugin to add `rel` (and `target`) to external links.
+
+## Contents
+
+*   [What is this?](#what-is-this)
+*   [When should I use this?](#when-should-i-use-this)
+*   [Install](#install)
+*   [Use](#use)
+*   [API](#api)
+    *   [`unified().use(rehypeExternalLinks[, options])`](#unifieduserehypeexternallinks-options)
+*   [Types](#types)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## What is this?
+
+This package is a [unified][] ([rehype][]) plugin to add `rel` (and `target`)
+attributes to external links.
+It is particularly useful when displaying user content on your reputable site,
+because users could link to disreputable sources (spam, scams, etc), as search
+engines and other bots will discredit your site for linking to them (or
+legitimize their sites).
+In short: linking to something signals trust, but you can’t trust users.
+This plugin adds certain `rel` attributes to prevent that from happening.
+
+**unified** is a project that transforms content with abstract syntax trees
+(ASTs).
+**rehype** adds support for HTML to unified.
+**hast** is the HTML AST that rehype uses.
+This is a rehype plugin that applies syntax highlighting to the AST.
+
+## When should I use this?
+
+This project is useful when you want to display user content from authors you
+don’t trust, such as articles or comments, on your website.
```

Comment:
This is for links specifically right?
What you're describing sounds like a sanitizer?

Reply:
Two main use cases of this plugin, that are sanitizer-like. The `nofollow` for external links, described as above, and the current `target=_blank`, which needs `noopener` and `noreferrer`.

Reply:
I'd consider specifying user generated links, rather than user generated content 🤔